ivre/utils.py

```diff
@@ -2213,3 +2214,101 @@ def deep_sort_dict_list(elt):
             value.sort()
         elif isinstance(value, dict):
             deep_sort_dict_list(value)
+
+
+def _extract_substr(ntlm_msg, offset, ln):
+    """
+    Extract the string at te given offset and of the given length from an
+    NTLM message
+    """
+    s, = struct.unpack('{}s'.format(ln), ntlm_msg[offset:offset + ln])
+    try:
+        res = s.decode('utf-16').encode()
+        if len(res) == ln // 2:
+            return res
+
+        LOGGER.warning("Size too small %r", s)
+        return None
+
+    except UnicodeDecodeError:
+        LOGGER.warning("Cannot decode %r", s)
+        return None
```

Comment:
Actually, the failure when `ntlm_msg[offset:offset + ln] is too short would occur in `struct.unpack()` so the test for `len(res) == ln // 2` comes too late. Also, you don't really need `struct` here...
```suggestion
    s = ntlm_msg[offset:offset + ln]
    if len(s) < ln:
        LOGGER.warning("Data too small at offset %d [%r, size %d]",
                       offset, ntlm_msg, msg)
        return None
    try:
        return s.decode('utf-16').encode()
    except UnicodeDecodeError:
        LOGGER.warning("Cannot decode %r", s)
        return None
```